.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,42 @@
+name: Docs build
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  docs_build:
+    name: Docs-Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout esp-protocols
+        uses: actions/checkout@master
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Generate docs
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install doxygen clang python3-pip
+          python -m pip install breathe recommonmark
+          cd $GITHUB_WORKSPACE/components/esp_modem/docs
+          ./generate_docs
+          cd $GITHUB_WORKSPACE/docs
+          touch .nojekyll
+          echo '<a href="esp_modem/index.html">esp-modem</a>' > index.html
+
+      - name: Commit generated docs
```

Comment:
There is a separate action for publishing on gh-pages in one step:
https://github.com/JamesIves/github-pages-deploy-action

It should make the job a bit cleaner.

Reply:
Thanks, used the two in one action!